provider/prometheus.go

```diff
@@ -0,0 +1,133 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"time"
+
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+
+	"github.com/evcc-io/evcc/provider/pipeline"
+	"github.com/evcc-io/evcc/util"
+	"github.com/prometheus/client_golang/api"
+)
+
+// Prometheus provider
+type Prometheus struct {
+	log         *util.Logger
+	client      api.Client
+	clientV1Api v1.API
+	query       string
+	scale       float64
+	timeout     time.Duration
+	pipeline    *pipeline.Pipeline
```

Comment:
The `pipeline` is never used below, or is it?

Reply:
Ups, left over by copy-pasting from mqtt. Removed, as this is only really useful when working with string values